src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -411,6 +419,11 @@ def _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=False):
         if for_scale_set:
             forbidden.append('os_disk_name')
```

Comment:
Is `os_disk_name` also forbidden by shared gallery image?

Reply:
```python
'SharedGalleryImage':{
    "osDisk": {
         ......
        "name": os_disk_name,
         ......
    },
    ......
}
```
No, it supports setting the `os_disk_name` for shared gallery image